src/store/juju/slice.ts

```diff
@@ -243,19 +242,20 @@ const slice = createSlice({
     },
     updateCrossModelQuery: (
       state,
-      { payload }: PayloadAction<CrossModelQueryFullResponse>,
+      { payload }: PayloadAction<CrossModelQueryResponse>,
     ) => {
-      // If "payload" is a string, it represents the error. In this case,
-      // "results" gets set to null and "errors" gets set to "payload".
-      state.crossModelQuery.results = isCrossModelQueryResponse(payload)
-        ? payload.results
-        : null;
-      state.crossModelQuery.errors =
-        isCrossModelQueryResponse(payload) && Object.keys(payload.errors).length
-          ? payload.errors
-          : typeof payload === "string"
-            ? payload
-            : null;
+      const hasErrors = !!Object.keys(payload.errors).length;
+      state.crossModelQuery.results = hasErrors ? null : payload.results;
+      state.crossModelQuery.errors = hasErrors ? payload.errors : null;
```

Comment:
```suggestion
      state.crossModelQuery.results = payload.results;
      state.crossModelQuery.errors = null;
```

See my other comment in model-poller.ts.